# ESTIMATION AND CONFIDENCE INTERVALS

In [7]:
import numpy as np
from scipy import stats

In [9]:
data = [1.13,1.55,1.43,0.92,1.25,1.36,1.32,0.85,1.07,1.43,1.20,1.33,1.18,1.22,1.29]

In [11]:
n =  len(data)

### 1. Calculate the smaple mean

In [19]:
sample_mean = np.mean(data)
sample_mean

1.2353333333333332

### Calculate the sample standard deviation:

In [79]:
sample_std = np.std(data, ddof=1)
sample_std

0.18909055527772428

### 3. Determine the t-score for a  99% confidence interval with 14 degrees of freedom:

In [27]:
t_score = stats.t.ppf(0.995,  df=n-1)
t_score

2.976842734370834

### 4.  Calculate the margin of error:

In [33]:
margin_of_error_t = t_score *(sample_std/np.sqrt(n))
margin_of_error_t

0.14533830778483664

### 5. Construct the confidence inverval:


In [39]:
confidence_interval_t = (sample_mean - margin_of_error_t, sample_mean + margin_of_error_t)
confidence_interval_t

(1.0899950255484965, 1.3806716411181699)

# part B: Using Know Population Standard Deviation

### 1.Use the know population standard deviation:

In [45]:
population_std = 0.2

### 2.Determine the Z-score for a 99% confidence interval:

In [55]:
z_score = stats.norm.ppf(0.995)
z_score

2.5758293035489004

### 3. calcualte the margin of error:

In [59]:
margin_of_error_z = z_score * (population_std/np.sqrt(n))
margin_of_error_z

0.13301525327090588

### 4.Construct the confidence interval:

In [67]:
confidence_interval_z = (sample_mean - margin_of_error_z, sample_mean + margin_of_error_z)
confidence_interval_z

(1.1023180800624273, 1.368348586604239)


## Final Answers	
### The 99% confidence interval using the sample standard deviation is {(1.09, 1.39)} million characters.
### The 99% confidence interval using the known population standard deviation is {(1.11, 1.37)} million characters.
